convert2rhel/systeminfo.py

```diff
@@ -152,49 +155,89 @@ def get_system_release_file_content():
-        content = self.system_release_file_content if not system_release_content else system_release_content
-        name = re.search(r"(.+?)\s?(?:release\s?)?\d", content).group(1)
-        return name
+    def parse_system_release_content(self, system_release_content=None):
+        """Parse the content of the system release string
 
-    def _get_system_version(self, system_release_content=None):
-        """Return a namedtuple with major and minor elements, both of an int type.
+        If system_release_content is not provided we use the content from the self.system_release_file_content.
 
-        Examples:
-        Oracle Linux Server release 7.8
-        CentOS Linux release 7.6.1810 (Core)
-        CentOS Linux release 8.1.1911 (Core)
+        :param system_release_content: The contents of the system_release file if needed.
+        :type system_release_content: str
+        :returns: A dictionary containing the system release information
+        :rtype: dict[str, str]
         """
+
         content = self.system_release_file_content if not system_release_content else system_release_content
-        match = re.search(r".+?(\d+)\.(\d+)\D?", content)
+
+        match = re.match(
+            # We assume that the /etc/system-release content follows the pattern:
+            # "<name> release <full_version> <Beta> (<dist_id>)"
+            # Here
+            # - <name>, parsed as a named group '(?P<name>.+?)',
+            #   is a non-empty string of arbitrary symbols.
+            # - "release ", parsed as (?:release\s)?,
+            #   is a literal string. It is optional, and it is dropped when parsed, so this group doesn't have a name.
+            # - <full_version>, parsed as a named group '(?P<full_version>[.\d]+)',
+            #   is a string of arbitrary length containing numbers and dots, for example 8.1.1911 or 7.9.
+            #   (For now we do not have examples with version strings containing letters or other symbols)
+            # - <Beta>, just the word Beta which may appear in some cases, parsed as '(?:\sBeta)?'.
+            # - <dist_id>, parsed as a named group (?P<dist_id>.+), is optional and when it is present, it must appear
+            #   in brackets. Thus, the named group is nested under an unnamed group which starts from a space and then
+            #   has brackets '(\s\( ...here goes the nested group... \))?'
+            #
+            # Example:
+            #
+            #   CentOS Stream release 8
+            #   <    name   > <      ><full_version>
+            #
+            #   CentOS Linux release 8.1.1911        (Core     )
+            #   <    name  > <      ><full_version><  <dist_id>>
+            #   CentOS Linux release 8.1.1911      Beta       (Core     )
+            #   <    name  > <      ><full_version><    ><  <dist_id>>
+            r"^(?P<name>.+?)\s(?:release\s)?(?P<full_version>[.\d]+)(?:\sBeta)?(\s\((?P<dist_id>.+)\))?$",
+            content,
```

Comment:
I get this warning locally

![image](https://github.com/oamg/convert2rhel/assets/6598829/8a7e4c27-c339-49c0-aa44-3e8ed386f0b1)

Reply:
Can you check now? I think it was caused by a naming collision so I simply changed `match` to `matched` for a variable name to not overlap with `re.match()`

Reply:
Seems like it's just the type hints from pylance. Since I don't know how to annotate in comments lets just ignore 

```suggestion
            # We assume that the /etc/system-release content follows the pattern:
            # "<name> release <full_version> <Beta> (<dist_id>)"
            # Here
            # - <name>, parsed as a named group '(?P<name>.+?)',
            #   is a non-empty string of arbitrary symbols.
            # - "release ", parsed as (?:release\s)?,
            #   is a literal string. It is optional, and it is dropped when parsed, so this group doesn't have a name.
            # - <full_version>, parsed as a named group '(?P<full_version>[.\d]+)',
            #   is a string of arbitrary length containing numbers and dots, for example 8.1.1911 or 7.9.
            #   (For now we do not have examples with version strings containing letters or other symbols)
            # - <Beta>, just the word Beta which may appear in some cases, parsed as '(?:\sBeta)?'.
            # - <dist_id>, parsed as a named group (?P<dist_id>.+), is optional and when it is present, it must appear
            #   in brackets. Thus, the named group is nested under an unnamed group which starts from a space and then
            #   has brackets '(\s\( ...here goes the nested group... \))?'
            #
            # Example:
            #
            #   CentOS Stream release 8
            #   <    name   > <      ><full_version>
            #
            #   CentOS Linux release 8.1.1911        (Core     )
            #   <    name  > <      ><full_version><  <dist_id>>
            #   CentOS Linux release 8.1.1911      Beta       (Core     )
            #   <    name  > <      ><full_version><    ><  <dist_id>>
            r"^(?P<name>.+?)\s(?:release\s)?(?P<full_version>[.\d]+)(?:\sBeta)?(\s\((?P<dist_id>.+)\))?$",
            content,  # type: ignore
```